mods/variations/init.lua

```diff
@@ -0,0 +1,99 @@
+variations = {}
+
+variations.variations = {
+	{
+		name = "brick";
+		description = "Brick";
+		texture = "0,0";
+	},
+	{
+		name = "big_tile";
+		description = "Big Tile";
+		texture = "1,0";
+	},
+	{
+		name = "tile";
+		description = "Tile";
+		texture = "2,0";
+	},
+	{
+		name = "tall_tile";
+		description = "Tall Tile";
+		texture = "0,1";
+	},
+	{
+		name = "small_brick";
+		description = "Small Brick";
+		texture = "1,1";
+	},
+	{
+		name = "small_tile";
+		description = "Small Tile";
+		texture = "2,1";
+	},
+	{
+		name = "horizontal_brick";
+		description = "Horizontal Brick";
+		texture = "0,2";
+	},
+	{
+		name = "cross_tile";
+		description = "Cross Tile";
+		texture = "1,2";
+	},
+	{
+		name = "spiral_tile";
+		description = "Spiral Tile";
+		texture = "2,2";
+	},
+}
+
+function variations.register_for_base(base_node, transparent)
+	local base_definition = minetest.registered_nodes[base_node]
+	for _, variation in ipairs(variations.variations) do
+		-- We get an iterator function over substrings split by :
+		local base_node_full_name = string.gmatch(base_node, "([^:]+)")
+		-- First call will give us the modname
+		-- local base_node_modname = base_node_full_name()
+		-- Unused variable warning...
+		base_node_full_name()
+		-- and the second will give us the node name
+		local base_node_name = base_node_full_name()
+		local variation_name = "variations:" .. base_node_name .. "_" .. variation.name
+		local variation_description = base_definition.description .. " " .. variation.description
+		local tiles = {"variations_" .. base_node_name .. ".png^[sheet:3x3:" .. variation.texture}
+		minetest.register_node(variation_name, {
+			description = variation_description,
+			tiles = tiles,
+			groups = base_definition.groups,
+			drawtype = base_definition.drawtype,
+			use_texture_alpha = transparent,
```

Comment:
split transparent up, as mentioned previously use_texture_alpha is deprecated, it can hold 3 different values now

Reply:
I also don't see any default values if this is not defined?
`transparent or base_definition.use_texture_alpha` ?

Reply:
nil == false?

Reply:
preferable if it copies the original base value i think.
Unless this is an issue that conflicts.